readme.md
## [![npm][npmjs-img]][npmjs-url] [![mit license][license-img]][license-url] [![build status][travis-img]][travis-url] [![coverage status][coveralls-img]][coveralls-url] [![deps status][daviddm-img]][daviddm-url]

> Common used char codes and their common used names as json object

## Install
```
npm i --save char-code-names
npm test
```


## Usage
> For more use-cases see the [tests](./test.js)

```js
var charCodeNames = require('char-code-names');

charCodeNames(42);   //=> ['star','asterisk'],
charCodeNames(44);   //=> ['comma']
charCodeNames('59'); //=> ['semicolon','semi-colon','semi colon']
charCodeNames();
//=> {
//  '32': ['space'],
//  '33': ['negate','exclam','exclamation','exclamation mark','exclam mark'],
//  '34': ['double','doublequote','double-quote','double quote'],
//  '35': ['hash','number','hash sign','number sign'],
//  '36': ['dollar','dollar sign']
//  ...
//}
```


## Author
**Charlike Mike Reagent**
+ [gratipay/tunnckoCore][author-gratipay]
+ [twitter/tunnckoCore][author-twitter]
+ [github/tunnckoCore][author-github]
+ [npmjs/tunnckoCore][author-npmjs]
+ [more ...][contrib-more]


## License [![MIT license][license-img]][license-url]
Copyright (c) 2015 [Charlike Mike Reagent][contrib-more], [contributors][contrib-graf].  
Released under the [`MIT`][license-url] license.


[npmjs-url]: http://npm.im/char-code-names
[npmjs-img]: https://img.shields.io/npm/v/char-code-names.svg?style=flat&label=char-code-names

[coveralls-url]: https://coveralls.io/r/tunnckoCore/char-code-names?branch=master
[coveralls-img]: https://img.shields.io/coveralls/tunnckoCore/char-code-names.svg?style=flat

[license-url]: https://github.com/tunnckoCore/char-code-names/blob/master/license.md
[license-img]: https://img.shields.io/badge/license-MIT-blue.svg?style=flat

[travis-url]: https://travis-ci.org/tunnckoCore/char-code-names
[travis-img]: https://img.shields.io/travis/tunnckoCore/char-code-names.svg?style=flat

[daviddm-url]: https://david-dm.org/tunnckoCore/char-code-names
[daviddm-img]: https://img.shields.io/david/dev/tunnckoCore/char-code-names.svg?style=flat

[author-gratipay]: https://gratipay.com/tunnckoCore
[author-twitter]: https://twitter.com/tunnckoCore
[author-github]: https://github.com/tunnckoCore
[author-npmjs]: https://npmjs.org/~tunnckocore

[contrib-more]: http://j.mp/1stW47C
[contrib-graf]: https://github.com/tunnckoCore/char-code-names/graphs/contributors

***

_Powered and automated by [kdf](https://github.com/tunnckoCore), March 17, 2015_

history.md


## v1.0.2 / March 21, 2015
- Release v1.0.2 / npm@v1.0.2
- migrate to `lab`

## v1.0.1 / March 17, 2015
- Release v1.0.1 / npm@v1.0.1
- update deps
- add more aliases
 
## v1.0.0 / March 17, 2015
- Release v1.0.0 / npm@v1.0.0
- fix badges
- add docs
- replace ava with mocha
- travis && coveralls

## v0.0.0 / March 17, 2015
- first commits / npm@v0.0.0

test.js
/**
 * char-code-names <https://github.com/tunnckoCore/char-code-names>
 *
 * Copyright (c) 2015 Charlike Mike Reagent, contributors.
 * Released under the MIT license.
 */

'use strict';

var lab = exports.lab = require('lab').script();
var assert = require('assert');
var forIn = require('for-in');
var charCodeNames = require('./index');

var describe = lab.describe;
var it = lab.it;

describe('char-code-names:', function() {
  it('length of char-code-names should be 32', function(done) {
    var i = 0;
    forIn(charCodeNames(), function forIn(val) {
      assert.ok(val);
      i++;
    });
    assert.strictEqual(i, 33);
    done();
  });
  it('values of object should be array', function(done) {
    forIn(charCodeNames(), function forIn(val) {
      assert.strictEqual(Array.isArray(val), true);
    });
    done();
  });
  it('keys of object should be number', function(done) {
    forIn(charCodeNames(), function forIn(val, key) {
      assert.strictEqual(typeof parseInt(key), 'number');
    });
    done();
  });
  it('char code `44` names length should be 1 and be `comma`', function(done) {
    assert.strictEqual(charCodeNames()['44'].length, 1);
    assert.strictEqual(charCodeNames()['44'][0], 'comma');
    done();
  });
  it('should get specific char code names', function(done) {
    assert.strictEqual(charCodeNames('44')[0], 'comma');
    assert.strictEqual(charCodeNames(44)[0], 'comma');
    done();
  });
});


index.js
/**
 * char-code-names <https://github.com/tunnckoCore/char-code-names>
 *
 * Copyright (c) 2015 Charlike Mike Reagent, contributors.
 * Released under the MIT license.
 */

'use strict';

var codeNames = {
  32: ['space'],
  33: ['negate', 'exclam', 'exclamation', 'exclamation mark', 'exclam mark'],
  34: ['double', 'doublequote', 'double-quote', 'double quote'],
  35: ['hash', 'number', 'hash sign', 'number sign'],
  36: ['dollar', 'dollar sign'],
  37: ['percent', 'percentage'],
  38: ['ampersand'],
  39: ['single', 'singlequote', 'single-quote', 'single quote'],
  40: ['left-parenthesis', 'left parenthesis', 'left-paren', 'left paren'],
  41: ['right-parenthesis', 'right parenthesis', 'right-paren', 'right paren'],
  42: ['star', 'asterisk'],
  43: ['plus', 'plus sign'],
  44: ['comma'],
  45: ['minus', 'dash', 'hyphen'],
  46: ['dot', 'period'],
  47: ['forward-slash', 'forwardslash', 'forward slash', 'slash'],
  58: ['colon'],
  59: ['semicolon', 'semi-colon', 'semi colon'],
  60: ['lower than', 'lowerthan', 'lower than sign', 'lower-than', 'lower-than sign', 'lt'],
  61: ['equal', 'equals'],
  62: ['greater than', 'greaterthan', 'greater than sign', 'greater-than', 'greater-than sign', 'gt'],
  63: ['question', 'question mark', 'q'],
  64: ['at', 'at sign'],
  91: ['left square bracket', 'left-square-bracket', 'left square', 'left-square', 'leftsquare', 'left-bracket', 'left bracket', 'leftbracket'],
  92: ['back-slash', 'backslash', 'back slash'],
  93: ['right square bracket', 'right-square-bracket', 'right square', 'right-square', 'rightsquare', 'right-bracket', 'right bracket', 'rightbracket'],
  94: ['caret'],
  95: ['horizontal', 'horizontal bar', 'underscore', 'lodash', 'horizontal line'],
  96: ['grave accent', 'accent'],
  123: ['left curly brace', 'left-curly-brace', 'left curly', 'left-curly', 'leftcurly', 'left-brace', 'left brace', 'leftbrace'],
  124: ['vertical', 'vertical bar', 'vertical line'],
  125: ['right curly brace', 'right-curly-brace', 'right curly', 'right-curly', 'rightcurly', 'right-brace', 'right brace', 'rightbrace'],
  126: ['tilde', 'tilda']
};

module.exports = function charCodeNames(code) {
  if (!code) {
    return codeNames;
  }
  return codeNames[String(code)];
};
